ggml/src/ggml-cann/kernels/get_row_float.cpp

```diff
@@ -41,16 +41,18 @@ class GET_ROW_F16 {
             ir = dr * op_block_idx + tails;
         }
 
-        input_gm.SetGlobalBuffer((__gm__ half *)input);
+        input_gm.SetGlobalBuffer((__gm__ ROW_TYPE *)input);
         indices_gm.SetGlobalBuffer((__gm__ int32_t *)indices);
         output_gm.SetGlobalBuffer((__gm__ float *)output);
 
-        uint64_t input_local_buffer_size = ((input_ne[0] * sizeof(half) + 31)
-                                             & ~31);
-        uint64_t output_local_buffer_size = ((input_ne[0] * sizeof(float) + 31)
-                                              & ~31);
-
-        local_buffer_elems = input_local_buffer_size / sizeof(half);
+        uint64_t input_local_buffer_size = ((input_ne[0] * sizeof(ROW_TYPE) + 31) & ~31);
+        uint64_t output_local_buffer_size = 0;
+        if constexpr (std::is_same<ROW_TYPE, half>::value) {
```

Comment:
This part of the code is redundant. Regardless of whether ROW_TYPE is half or float, the execution logic is the same, so there is no need to make a distinction.

In addition, can you write memory alignment as an inline function? This way the code looks cleaner